### PR TITLE
some gen 9 move anims + fickle beam rework

### PR DESCRIPTION
**Pull request by TheTrueSadfish**
- Added new animations for these moves: SPIN_OUT, MORTAL_SPIN, FILLET_AWAY, FLOWER_TRICK, MAKE_IT_RAIN, SHED_TAIL, HYPER_DRILL, TWIN_BEAM, COMEUPPANCE, BLOOD_MOON, FICKLE_BEAM, THUNDERCLAP, HARD_PRESS, DRAGON_CHEER, MALIGNANT_CHAIN

- Fixed Raging Bull's animation to include the screen break like Brick Break

- Changed Fickle Beam's effect to be like magnitude to include the battle message "is going all out for this attack"

also a save to test em all: [pokeemerald.zip](https://github.com/user-attachments/files/16604417/pokeemerald.zip)

purple chains by ogwon on discord
blood moon and red explosions by me (but they're just recolors lol)
beam by me/livra on discord

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -17132,3 +17132,18 @@ void BS_DamageToQuarterTargetHP(void)
...
+    {
+        gBattleStruct->fickleBeamBasePower = 160;
+        gBattlescriptCurrInstr = BattleScript_FickleBeamDoubled;
+    }
+    else
+    {
+        gBattleStruct->fickleBeamBasePower = 80;
+        gBattlescriptCurrInstr = cmd->nextInstr;
+    }
+}
```
Comment (AlexOn1ine) [inline src/battle_script_commands.c:17149]: This isn't needed because it is already handled in `battle_util.c`
```
    case EFFECT_FICKLE_BEAM:
        if (RandomPercentage(RNG_FICKLE_BEAM, 30))
            basePower *= 2;
        break;
```

File: src/battle_util.c [hunk 2]
```diff
@@ -9073,8 +9073,7 @@ static inline u32 CalcMoveBasePower(u32 move, u32 battlerAtk, u32 battlerDef, u3
         basePower = (basePower > 350) ? 350 : basePower;
         break;
     case EFFECT_FICKLE_BEAM:
-        if (RandomPercentage(RNG_FICKLE_BEAM, 30))
-            basePower *= 2;
+        basePower = gBattleStruct->fickleBeamBasePower;
```
Comment (AlexOn1ine) [inline src/battle_util.c:9076]: what is the reason for this change?

**Comment by hedara90** [comment]
I'll look at the animation code tomorrow.
But looking at the animations in game they look good.
Just 2 comments about that.
pokeemerald-5
I think the spin anim sprite should use a more purple palette.
pokeemerald-4
Should the thunderbolt reach further down so it doesn't stop halfway down the sprite?

**Comment by TheTrueSadfish** [comment]
pokeemerald
pokeemerald1
these should hopefully look better

**Comment by hedara90** [comment]
pokeemerald-6
I meant more the spinny bits of Mortal Spin, here I just slapped `ANIM_TAG_POISON_BUBBLE` into the palette since it was already loaded.

File: data/battle_anim_scripts.s [hunk 3]
```diff
@@ -17013,44 +17062,626 @@ Move_AXE_KICK::
...
+	createsprite gCrushGripExplosionTemplate, ANIM_TARGET, 0, 0, 0, 1, 1
+	delay 3
+	playsewithpan SE_M_EXPLOSION, SOUND_PAN_ATTACKER
+	createsprite gCrushGripExplosionTemplate, ANIM_TARGET, 0, 0x18, 0xffe8, 1, 1
+	delay 3
+	playsewithpan SE_M_EXPLOSION, SOUND_PAN_ATTACKER
+	createsprite gCrushGripExplosionTemplate, ANIM_TARGET, 0, 0xfff0, 16, 1, 1
+	delay 3
+	playsewithpan SE_M_EXPLOSION, SOUND_PAN_ATTACKER
+	createsprite gCrushGripExplosionTemplate, ANIM_TARGET, 0, 0xffe8, -12, 1, 1
```
Comment (hedara90) [inline data/battle_anim_scripts.s:17595]: Unless there's a specific reason to use hex numbers, decimal numbers are preferred.
`0xffe8` -> `-24`, `0xfff0` -> `-16`, `0x18`->`24`

File: data/battle_anim_scripts.s [hunk 4]
```diff
@@ -17013,44 +17062,626 @@ Move_AXE_KICK::
...
+	loadspritegfx ANIM_TAG_PURPLE_CHAIN
+	loadspritegfx ANIM_TAG_POISON_BUBBLE
+	loopsewithpan SE_M_SCRATCH, SOUND_PAN_TARGET, 6, 2
+	createsprite gChainBindingSpriteTemplate, ANIM_TARGET, 4, 0, 16, 0, 1
+	delay 7
+	createsprite gChainBindingSpriteTemplate, ANIM_TARGET, 2, 0, 8, 1, 1
+	delay 3
+	createvisualtask AnimTask_ShakeMon2, 2, ANIM_TARGET, 2, 0, 8, 1
+	delay 20
+	setarg 7, 0xFFFF
```
Comment (hedara90) [inline data/battle_anim_scripts.s:17655]: Unless there's a specific reason to use hex numbers, decimal numbers are preferred.
`0xFFFF` -> `-1`, usage of `setarg 7, ####` is a bit inconsistent, if makes more sense to have `0xFFFF` then use that.
Maybe someone who knows more about the anim system could weigh in?

File: data/battle_anim_scripts.s [hunk 5]
```diff
@@ -17013,44 +17062,626 @@ Move_AXE_KICK::
...
+	waitforvisualfinish
+	end
+
+Move_HARD_PRESS::
+	loadspritegfx ANIM_TAG_EXPLOSION
+	loadspritegfx ANIM_TAG_ACUPRESSURE
+	loadspritegfx ANIM_TAG_PURPLE_HAND_OUTLINE
+	setalpha 15, 0
+	call SetSteelBeamBackground
+	createvisualtask AnimTask_CompressTargetHorizontally, 2
```
Comment (hedara90) [inline data/battle_anim_scripts.s:17585]: The compression could have less duration.
Duration = 32

https://github.com/user-attachments/assets/658c4689-c4a2-43f2-9631-fa7756543854

Duration = 64 (current)

https://github.com/user-attachments/assets/37a9c87c-0750-4b52-8971-3ef574854e96

**Review by hedara90 — CHANGES_REQUESTED** [review]
Code style changes and comment on Hard Press

File: data/battle_anim_scripts.s (hunk 5 above)
Comment (TheTrueSadfish) [reply data/battle_anim_scripts.s:17585]: oh how did you change the duration

**Comment by TheTrueSadfish** [comment]
mortalspin
did a blendparticle instead since i thought it looked nicer

File: data/battle_anim_scripts.s (hunk 5 above)
Comment (hedara90) [reply data/battle_anim_scripts.s:17585]: I changed the 64 in `sCompressTargetHorizontallyAffineAnimCmds` to 32, you'd probably have to make a new `AnimTask` and `AffineAnimCommand`, but they are all right next to each other in `src/battle_anim_effects_1.c`.

**Review by hedara90 — APPROVED** [review]
